### DISCUSSION TITLE
Creating DOCX Files from Within Freeplane (Similar to PNG) and an Idea for Exporting Freeplane Files with Linked Content.

### DISCUSSION THREAD

**Discussion by aaa1386** [freeplane/freeplane#2134] [Ideas]
Freeplane allows users to create a PNG file directly from within the software. For example, I created a PNG file from a node named "Books," and the file was automatically saved in a folder with the same name as the map. In the image below, you can see that I have selected node 1111 and I am creating a PNG file from within Freeplane.

![image](https://github.com/user-attachments/assets/35a55c33-83e6-4302-ad21-056dfdf99428)

________________________________________
 **_Idea 1: Direct Creation of DOCX Files from Nodes_**
Many times, users deal with long texts that they want to associate with their Freeplane maps. When the text is too long, users often prefer to manage it in a Word document (DOCX).
 Proposal : Introduce a feature that enables users to directly create a DOCX file from a node in the map, just as they can create a PNG file.
**Benefits**
•	**Organized File Management:**
All DOCX files can be stored in a folder located next to the MM file. Each mind map (MM) will have a dedicated folder where related texts and images are stored, keeping everything organized and easily accessible.
•	**Avoiding Map Overcrowding:**
For lengthy texts, using DOCX files is more efficient. This prevents the mind map from becoming cluttered, while still allowing easy access to the content, as links to the text files are included in the relevant nodes of the map.
•	**Seamless Integration with Word Files:**
Linking a Word file to a node can be straightforward. The user can issue a command within the program to create a DOCX file directly, which will automatically be linked to the selected node in the map.
•	**Addressing Broken Links:**
When linked text files come from different locations, moving the MM file or its linked files can break the connections. To avoid this, all linked files should be stored in the same folder as the MM file. This ensures that links remain intact and the files can be accessed without any issues.

Additionally:
•	Users can link the DOCX file to the corresponding node in the map.
•	The DOCX file can be opened and edited directly from within Freeplane.
•	This provides flexibility for handling long texts while keeping the mind map organized.
________________________________________
 **_Idea 2: Seamless File Management and Export_**
One major challenge with Freeplane is that when transferring a map file (MM) to another computer or hard drive, the associated PNG, DOCX, or other linked files are often lost.
**Proposal**:
Implement a feature inspired by the Anki software:
•	In Anki, when users export a batch of cards that include linked media (e.g., images, audio files), the software provides an option to include those media files in the export package.
 **Suggested Approach**
If Freeplane adopts a similar system, users could export their maps along with all linked files (texts, images, or other media) in a single package.
 **Benefits**
•	Ensures maps and their associated content remain functional and accessible regardless of location.
•	Maintains the integrity of linked files, preventing loss during transfers.

![image](https://github.com/user-attachments/assets/b70d5efc-65ad-4aa3-85aa-d904dbd555e9)

**Comment by quickfold** [comment]
Freeplane already has several export options, and although .docx isn't one of them, .doc is, as well as other text formats.
